tests/tensorflow/test_tensorflow_autolog.py

```diff
@@ -0,0 +1,213 @@
+# pep8: disable=E501
+
+from __future__ import print_function
+
+import collections
+import shutil
+import pytest
+import tempfile
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow.python.keras import layers
+
+import mlflow
+import mlflow.tensorflow
+
+SavedModelInfo = collections.namedtuple(
+        "SavedModelInfo",
+        ["path", "meta_graph_tags", "signature_def_key", "inference_df", "expected_results_df"])
+
+
+@pytest.fixture
+def tf_keras_random_data():
+    mlflow.tensorflow.autolog(metrics_every_n_steps=2)
+    def random_one_hot_labels(shape):
+        n, n_class = shape
+        classes = np.random.randint(0, n_class, n)
+        labels = np.zeros((n, n_class))
+        labels[np.arange(n), classes] = 1
+        return labels
+
+    data = np.random.random((1000, 32))
+    labels = random_one_hot_labels((1000, 10))
+
+    model = tf.keras.Sequential()
+
+    model.add(layers.Dense(64, activation='relu'))
+    model.add(layers.Dense(64, activation='relu'))
+    model.add(layers.Dense(10, activation='softmax'))
+
+    model.compile(optimizer=tf.train.AdamOptimizer(0.001),
+                  loss='categorical_crossentropy',
+                  metrics=['accuracy'])
+    run = mlflow.start_run()
+    model.fit(data, labels, epochs=10)
+    mlflow.end_run()
+    return run.info.run_id
+
+
+@pytest.fixture
+def get_tf_keras_random_data_run_instance(tf_keras_random_data):
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(tf_keras_random_data)
+    return run
+
+
+@pytest.mark.small
+def test_tf_keras_autolog_logs_epoch_acc(get_tf_keras_random_data_run_instance):
+    assert 'epoch_acc' in get_tf_keras_random_data_run_instance.data.metrics
+
+
+@pytest.mark.small
+def test_tf_keras_autolog_logs_epoch_loss(get_tf_keras_random_data_run_instance):
+    assert 'epoch_loss' in get_tf_keras_random_data_run_instance.data.metrics
+
+
+@pytest.mark.small
+def test_tf_keras_autolog_logs_optimizer_name(get_tf_keras_random_data_run_instance):
+    assert 'optimizer_name' in get_tf_keras_random_data_run_instance.data.params
+
+
+@pytest.mark.small
+def test_tf_keras_autolog_logs_model_summary(get_tf_keras_random_data_run_instance):
+    assert 'summary' in get_tf_keras_random_data_run_instance.data.tags
+    assert get_tf_keras_random_data_run_instance.data.tags['summary'] is not ''
+
+
+@pytest.mark.large
```

Comment:
Probably don't need to mark this as large since it depends on the same fixture as the small tests (i.e. it won't be any slower than them). If the model-training fixture (`get_tf_keras_random_data_run_instance`) is slow though, we should either find a way to make it fast or mark all the tests as large.

Reply:
Thanks for the heads up. I've left the `large` decorator on the Estimator test because the associated fixture for that one is pretty slow.